### ISSUE TITLE
Application Crash event fired during the srartup, from module nt.dll

### ISSUE THREAD

**Issue by laolarou726** [AvaloniaUI/Avalonia#12926]
**Describe the bug**

Avalonia application failed to startup with error in nt.dll.

**Screenshots**

<img width="631" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/25716486/2a84174b-1c4c-4679-93b5-d1ab3b97b393">
<img width="1298" alt="image" src="https://github.com/AvaloniaUI/Avalonia/assets/25716486/e7f2ea56-2252-4c71-90a0-b0cf559d3367">

**Desktop (please complete the following information):**

- OS: Windows 11 22H2
- Version: 11.0.2

**Additional context**

Files from events viewer:
[Events.zip](https://github.com/AvaloniaUI/Avalonia/files/12644718/Events.zip)

**Comment by timunie** [comment]
please double-check using 11.0.4 or even better nightly builds. If you can reproduce using latest version, please file a minimal sample for us to check. 

Also can you please translate the response?

**Comment by laolarou726** [comment]
> please double-check using 11.0.4 or even better nightly builds. If you can reproduce using latest version, please file a minimal sample for us to check. 
> 
> 
> 
> Also can you please translate the response? 

Sorry about that, those error files are from one of our client. So maybe it is hard for me to translate it to English. But I'll give a try with the latest nightly version. I've already tried 11.0.4 but the issue is still exists. I think that might related to the executable trimming. When I disable the trimming everything works perfect.

**Comment by timunie** [comment]
trimming requires CompiledBindings. Do you have them enabled everywhere? Make also sure any other lib is not using reflection or you have a root file handling these types.

**Comment by laolarou726** [comment]
> trimming requires CompiledBindings. Do you have them enabled everywhere? Make also sure any other lib is not using reflection or you have a root file handling these types. 

Yes. I enabled CompiledBinding in the csproj. Actually, the trimmed binary works on most of the clients' computer except few of them. They all have the same crash event.

**Comment by timunie** [comment]
I found this: https://answers.microsoft.com/en-us/windows/forum/all/app-crash-with-ntdlldll/9aa59f80-99a1-4f3c-b2f0-7eb72df05b2a

Other than that, I wonder how we can fix that issue on our side, As the issue doesn't seem to be reproducible.
